src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/scalable/ScoreVariantAnnotations.java

```diff
@@ -0,0 +1,322 @@
+            if (scoresIterator.hasNext()) {
+                throw new IllegalStateException("Traversals of scores and variants " +
+                        "(or alleles, in allele-specific mode) were not correctly synchronized.");
+            }
+            if (vcfWriter != null) {
+                vcfWriter.close();
+            }
+        }
+    }
+
+    private void readAnnotationsAndWriteScoresToHDF5() {
+        final List<String> annotationNames = LabeledVariantAnnotationsData.readAnnotationNames(outputAnnotationsFile);
+        final List<Boolean> isSNP = LabeledVariantAnnotationsData.readLabel(outputAnnotationsFile, "snp");
+        final double[][] allAnnotations = LabeledVariantAnnotationsData.readAnnotations(outputAnnotationsFile);
+        final int numAll = allAnnotations.length;
+        final List<Double> allScores = new ArrayList<>(Collections.nCopies(numAll, Double.NaN));
+        if (variantTypesToExtract.contains(VariantType.SNP)) {
+            logger.info("Scoring SNP variants...");
+            scoreVariantTypeAndSetElementsOfAllScores(annotationNames, allAnnotations, isSNP, snpScorer, allScores);
+        }
+        if (variantTypesToExtract.contains(VariantType.INDEL)) {
+            logger.info("Scoring INDEL variants...");
+            final List<Boolean> isIndel = isSNP.stream().map(x -> !x).collect(Collectors.toList());
+            scoreVariantTypeAndSetElementsOfAllScores(annotationNames, allAnnotations, isIndel, indelScorer, allScores);
+        }
+        VariantAnnotationsScorer.writeScores(outputScoresFile, Doubles.toArray(allScores));
+        logger.info(String.format("Scores written to %s.", outputScoresFile.getAbsolutePath()));
+    }
+
+    private static void scoreVariantTypeAndSetElementsOfAllScores(final List<String> annotationNames,
+                                                                  final double[][] allAnnotations,
+                                                                  final List<Boolean> isVariantType,
+                                                                  final VariantAnnotationsScorer variantTypeScorer,
+                                                                  final List<Double> allScores) {
+        final File variantTypeAnnotationsFile = LabeledVariantAnnotationsData.subsetAnnotationsToTemporaryFile(annotationNames, allAnnotations, isVariantType);
+        final File variantTypeScoresFile = IOUtils.createTempFile("temp", ".scores.hdf5");
+        variantTypeScorer.score(variantTypeAnnotationsFile, variantTypeScoresFile);
+        final double[] variantTypeScores = VariantAnnotationsScorer.readScores(variantTypeScoresFile);
+        final Iterator<Double> variantTypeScoresIterator = Arrays.stream(variantTypeScores).iterator();
+        IntStream.range(0, allScores.size()).filter(isVariantType::get).forEach(i -> allScores.set(i, variantTypeScoresIterator.next()));
+    }
+
+    @Override
+    void writeExtractedVariantToVCF(final VariantContext vc,
```

Comment:
@meganshand you'll want to modify this VCF-writing method to perform hard filtering as desired. Currently, we don't actually do any hard filtering; we just take the appropriate score(s) for each variant from a global iterator, use a truth-sensitivity converter (which is built from the appropriate set of truth scores) to convert the score (or the maximum of scores, in AS mode) to a sensitivity, and then write both to the VCF.

Seems like you could add whatever tool-level parameters you'll need to specify the cutoff calculation, and then store the resulting threshold globally. And perhaps do this separately for SNP/INDEL.